### PR TITLE
Es5

### PR DESCRIPTION
**Pull request by bermanjosh**
PR for #157

### CODE REVIEW

**Comment by MichaelXavier** [comment]
I'd recommend updating the test matrix to at least include ES 2.4.2 for one of the GHC versions. Probably would be good to start with 1 until its passing.

File: .travis.yml [hunk 1]
```diff
@@ -4,19 +4,19 @@ addons:
   apt:
     packages:
       - libgmp-dev
-      - openjdk-6-jre
+      - oracle-java8-installer
```
Comment (bitemyapp) [inline .travis.yml]: Was this to diagnose/eliminate a runtime problem? I'd like to understand the motivation.

Comment (bermanjosh) [inline .travis.yml]: Elastic 5.x needs Java 8 [1]
Travis only supports Oracle JDK 8 [2]

[1] - https://www.elastic.co/guide/en/elasticsearch/reference/current/setup.html#jvm-version
[2] - https://docs.travis-ci.com/user/ci-environment/#JDK

Comment (bitemyapp) [inline .travis.yml]: Thanks!

**Comment by bermanjosh** [comment]
@bitemyapp @MichaelXavier 
Hey - the merge is clean and tests pass. What's next?

**Comment by bitemyapp** [comment]
Do we block on explicit ES2 compat?

Also I'll be flying out to Colorado for the LC Winter Retreat, so my availability is going to be spotty for the next week.

**Comment by bermanjosh** [comment]
Personally, I don't think it's worth maintaining. If no one was clamoring for it to this point, and we've now eclipsed it, I don't see a reason to develop for it. 

I can run the V5 modules against ES2 locally and see if / how badly they fail. We can then add some appropriate note to the readme based on that.

**Comment by bermanjosh** [comment]
@bitemyapp Oh, also - enjoy! GL on the talk.

**Comment by bitemyapp** [comment]
I'm fine lazily evaluating on ES2 support but somehow thought it was something that would be nice for @MichaelXavier - I may be mis-remembering.

**Comment by bermanjosh** [comment]
I thought @MichaelXavier meant just to see. 

My concern is if it fails (which isn't totally unlikely)? If we're not committing to fixing those failures, won't that fail every subsequent commit?

**Comment by bitemyapp** [comment]
>I thought @MichaelXavier meant just to see.

That tickles a part of my memory yes.

>My concern is if it fails (which isn't totally unlikely)? If we're not committing to fixing those failures, won't that fail every subsequent commit?

We won't block on that then.

**Comment by MichaelXavier** [comment]
Sorry for the silence. I'm on 1.7 right now and I think the official path is 1 -> 2 -> 5. However, I'm not going to tax your efforts @bermanjosh. My time on 2 will probably be brief. So I say don't let 2 hold up 5. If it works for 2: great. If it doesn't and I need it for upgrade, 5 will serve as a model for 2 and I may contribute 2 support back if I end up needing it.

**Comment by bermanjosh** [comment]
@MichaelXavier Ok, thanks. 

I just pushed an update to include tests for 2 against both the V1 and V5 modules. When I tested them locally, I got 14 and 15 failures. So, basically, the situation hasn't really changed. Es2 might work, but nothing's guaranteed.

**Comment by MichaelXavier** [comment]
We may want to disable that part of the test matrix before it goes to master so we have a green suite there.

**Comment by bermanjosh** [comment]
I think I did. Though it looks like it's hiding the fact I put the links in incorrectly.

File: .travis.yml [hunk 2]
```diff
@@ -4,19 +4,26 @@ addons:
...
+   - GHCVER=7.10 ESVER=1.5.2 ESFLAG=ES1 DLINK=https://download.elastic.co/elasticsearch/elasticsearch/elasticsearch
+   - GHCVER=7.10 ESVER=1.6.0 ESFLAG=ES1 DLINK=https://download.elastic.co/elasticsearch/elasticsearch/elasticsearch
+   - GHCVER=7.10 ESVER=1.7.2 ESFLAG=ES1 DLINK=https://download.elastic.co/elasticsearch/elasticsearch/elasticsearch
+   - GHCVER=8.0  ESVER=1.7.2 ESFLAG=ES1 DLINK=https://download.elastic.co/elasticsearch/elasticsearch/elasticsearch
+   - GHCVER=8.0  ESVER=2.4.2 ESFLAG=ES1 DLINK=https://download.elastic.co/elasticsearch/release/org/elasticsearch/distribution/tar/elasticsearch/2.4.2/elasticsearch
+   - GHCVER=8.0  ESVER=2.4.2 ESFLAG=ES5 DLINK=https://download.elastic.co/elasticsearch/release/org/elasticsearch/distribution/tar/elasticsearch/2.4.2/elasticsearch
+   - GHCVER=8.0  ESVER=5.0.2 ESFLAG=ES5 DLINK=https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch ES_JAVA_OPTS="-Xms500m -Xmx750m"
+
+matrix:
+  allow_failures:
```
Comment (bermanjosh) [inline .travis.yml]: @bitemyapp @MichaelXavier 
Any idea why this isn't allowing the es2 failures through?

Comment (MichaelXavier) [inline .travis.yml]: Going by the docs this seems like this should work. I assume its just travis not working as indicated.

Comment (bermanjosh) [inline .travis.yml]: So should I take the tests out?

Comment (MichaelXavier) [inline .travis.yml]: I'd say sure, or comment them out. I imagine if I'm working on v2 backports I'll create a branch, reenable the tests and work from there.

Comment (bermanjosh) [inline .travis.yml]: @MichaelXavier

**Comment by bermanjosh** [comment]
@bitemyapp I'll try to bring in @MichaelXavier's aeson work over the weekend, or early next week.

**Comment by bermanjosh** [comment]
@bitemyapp What's next?

**Comment by bitemyapp** [comment]
@bermanjosh @MichaelXavier merged so we can just make a mess / fix things in master. Next release should include the split modules. As I understand it, we're only committing to V1/V5 for now unless someone steps up for V2.

/cc @MHova

**Pull request merged**